### PR TITLE
zsh auto completion

### PR DESCRIPTION
**Pull request by trevorlinton**

### CODE REVIEW

File: lib/completion.js [hunk 1]
```diff
@@ -79,8 +79,10 @@ module.exports = function completion (yargs, usage, command) {
 
   // generate the completion script to add to your .bashrc.
   self.generateCompletionScript = function generateCompletionScript ($0, cmd) {
+    let zshShell = process.env.SHELL && process.env.SHELL.indexOf('zsh') !== -1
```
Comment (bcoe) [inline lib/completion.js]: I believe this could be `const` and not `let`.

File: completion.zsh.hbs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# yargs command completion script
+#
+# Installation: {{app_path}} {{completion_command}} >> ~/.zshrc
+#    or {{app_path}} {{completion_command}} >> ~/.zsh_profile on OSX.
+#
+_{{app_name}}_yargs_completions()
+{
+  local reply
+  local si=$IFS
+  IFS=$'\n' reply=($(COMP_CWORD="$((CURRENT-1))" COMP_LINE="$BUFFER" COMP_POINT="$CURSOR" {{app_path}} --get-yargs-completions "$\{words[@]\}"))
```
Comment (bcoe) [inline completion.zsh.hbs]: interesting, at the end of the day this isn't too far off bash completion.

File: test/completion.js [hunk 3]
```diff
@@ -196,10 +196,21 @@ describe('Completion', () => {
   })
 
   describe('generateCompletionScript()', () => {
-    it('replaces application variable with $0 in script', () => {
+    it('replaces application variable with $0 in script (bash)', () => {
+      process.env.SHELL = '/bin/bash'
```
Comment (bcoe) [inline test/completion.js]: I like this approach 👍

File: completion.zsh.hbs (hunk 2 above)
Comment (trevorlinton) [reply completion.zsh.hbs]: yeah, not to much of a difference really.

File: lib/completion.js (hunk 1 above)
Comment (trevorlinton) [reply lib/completion.js]: fixed.

**Pull request merged**